openeo/rest/connection.py

```diff
@@ -1285,17 +1285,23 @@ def _config_log(message):
 
     if not url:
         raise OpenEoClientException("No openEO back-end URL given or known to connect to.")
-    connection = Connection(url, session=session, default_timeout=default_timeout)
-
-    auth_type = auth_type.lower() if isinstance(auth_type, str) else auth_type
-    if auth_type in {None, False, 'null', 'none'}:
-        pass
-    elif auth_type == "basic":
-        connection.authenticate_basic(**(auth_options or {}))
-    elif auth_type in {"oidc", "openid"}:
-        connection.authenticate_oidc(**(auth_options or {}))
-    else:
-        raise ValueError("Unknown auth type {a!r}".format(a=auth_type))
+    
+    try:
+        connection = Connection(url, session=session, default_timeout=default_timeout)
+        auth_type = auth_type.lower() if isinstance(auth_type, str) else auth_type
+        if auth_type in {None, False, 'null', 'none'}:
+            pass
+        elif auth_type == "basic":
+            connection.authenticate_basic(**(auth_options or {}))
+        elif auth_type in {"oidc", "openid"}:
+            connection.authenticate_oidc(**(auth_options or {}))
+        else:
+            raise ValueError("Unknown auth type {a!r}".format(a=auth_type))
+    except Exception as e:
```

Comment:
Switching to `LocalConnection`  through a `try-except` doesn't look very clean.

In this experimental phase of this feature, I would avoid touching `rest/connection.py` and related and work in an isolated submodule

Reply:
Ok, I will move to a different module instead of having it there. From `openeo/rest/connection.py` to `openeo/local/connection.py`

Reply:
Removed the new local parts from `openeo/rest/connection.py` and introduced `openeo/local/connection.py` in this commit: https://github.com/Open-EO/openeo-python-client/pull/338/commits/20647753e5fd0df25d26e3251c9243142d7a4470